src/Smalot/PdfParser/Document.php

```diff
@@ -149,6 +151,43 @@ protected function buildDetails()
             $details['Pages'] = 0;
         }
 
+        // Decode and repair encoded document properties
+        foreach ($details as $key => $value) {
+            if (\is_string($value)) {
+
+                // If the string is already UTF-8 encoded, that means we only
+                // need to repair Adobe's ham-fisted insertion of line-feeds
+                // every ~127 characters, which doesn't seem to be multi-byte
+                // safe
+                if (mb_check_encoding($value, 'UTF-8')) {
+
+                    $value = str_replace("\x5c\x0d", '', $value);
+
+                    while (preg_match("/\x5c\x5c\xe0([\xb4-\xb8])(.)/", $value, $match)) {
+                        $diff = (\ord($match[1]) - 182) * 64;
+                        $newbyte = PDFDocEncoding::convertPDFDoc2UTF8(\chr(\ord($match[2]) + $diff));
+                        $value = preg_replace("/\x5c\x5c\xe0".$match[1].$match[2]."/", $newbyte, $value);
+                    }
+
+                    while (preg_match("/(.)\x9c\xe0([\xb3-\xb7])/", $value, $match)) {
```

Comment:
Please add a short 1-line comment for each while-loop in the code about what they do.

Reply:
Helpful comments now, thanks.